src/Components/Admin/EnrollPatient.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (mode == 'add')
+      handleAdd();
+    if (mode == 'edit')
+      handleEdit();
+  };
+
+
+
+  const handleAdd = async () => {
+    try {
+      const requestBody = {
+        "user": {
+          "email": formData.email,
+          "password": formData.password
+        }
+      }
+      const responseForUser = await fetch('http://localhost:3000/users', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': localStorage.getItem('token') as string
+        },
+        body: JSON.stringify(requestBody),
+      });
+      const dataForUsers: ApiResponseForUsers = await responseForUser.json();
+      if(dataForUsers.status.errors && dataForUsers.status.errors.length>0){
+        dataForUsers.status.errors.forEach((ele:string)=>{
+          toast.success(ele)
+        })
+      }
+
+      const requestBodyForUserDetails = {
+        "user_detail": {
+          "first_name": formData.first_name,
+          "last_name": formData.last_name,
+          "date_of_birth": formData.date_of_birth,
+          "gender": formData.gender,
+          "phone_no": formData.phone_no,
+          "user_id": dataForUsers.status.data.id,
+          "role_id": 3
+        }
+      }
+
+      const responseForUserDetails = await fetch('http://localhost:3000/user_details', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': localStorage.getItem('token') as string
+        },
+        body: JSON.stringify(requestBodyForUserDetails),
+      });
+      const dataForUserDetails = await responseForUserDetails.json();
+      onSuccess(dataForUserDetails);
+      toast.success('User Added successful!');
+      onClose();
+    } catch (error) {
+      console.error('Error adding Patient:', error);
+    }
+  }
+
+  const handleEdit = async () => {
+    try {
+      const requestBodyForUserDetails = {
+        "user_detail": {
+          "first_name": formData.first_name,
+          "last_name": formData.last_name,
+          "date_of_birth": formData.date_of_birth,
+          "gender": formData.gender,
+          "phone_no": formData.phone_no,
+          "user_id": rowData.user_id,
+          "role_id": 3
+        }
+      }
+      const responseForUserDetails = await fetch(`http://localhost:3000/user_details/${rowData.user_id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': localStorage.getItem('token') as string
+        },
+
+        body: JSON.stringify(requestBodyForUserDetails),
+      });
+      onSuccess(responseForUserDetails);
+      toast.success('User Updated successful!');
+      onClose();
+    } catch (error) {
+      console.error('Error adding Patient:', error);
+    }
+  }
+
+
+  return (
+
+    <div className="modal" style={{ display: 'block' }}>
```

Comment:
move this in separate component

Reply:
Its model , so there is need to move this ?